packages/core/src/queries/roles.ts

```diff
@@ -0,0 +1,20 @@
+import { Roles, Role } from '@logto/schemas';
+import { sql } from 'slonik';
+
+import pool from '@/database/pool';
+import { convertToIdentifiers } from '@/database/utils';
+
+const { table, fields } = convertToIdentifiers(Roles);
+
+export const findAllRoles = async () =>
+  pool.many<Role>(sql`
+    select ${sql.join(Object.values(fields), sql`, `)}
+    from ${table}
+  `);
+
+export const findRolesByRoleName = async (roleNames: string[]) =>
+  pool.any<Role>(sql`
```

Comment:
`.many()`?

Reply:
If a single invalid `roleNames` are provided slonik will throw an empty resource not found error, which will lose context of the requested `roleNames`.  So use `any` here, and handle the empty array on the upper scope.